### PR TITLE
Combine redundant/nested FormField Boxes for readability

### PR DESCRIPTION
**Pull request by halocline**
Signed-off-by: Matt Glissmann <mdglissmann@gmail.com>

#### What does this PR do?
Combines two `Box`es used in implementation of `FormField` content. The intention is 1) to consolidate `Box` properties which are currently spread across 2 instances, but don't appear both to be needed, 2) to improve readability and make it easier to reason about the implementation, 3) remove extraneous divs and classes in DOM.

#### Where should the reviewer start?
FormField.js

#### What testing has been done on this PR?
Storybook

#### How should this be manually tested?
Storybook

#### Any background context you want to provide?
Breaking https://github.com/grommet/grommet/pull/4140 into 2. This is 1 of 2.

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
No.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

File: src/js/components/FormField/FormField.js [hunk 1]
```diff
@@ -234,7 +234,6 @@ const FormField = forwardRef(
         contentProps.background = formFieldTheme.disabled.background;
       }
     }
-    contents = <Box {...contentProps}>{contents}</Box>;
```
[the remark this answers is not in the document]
Comment (halocline) [reply src/js/components/FormField/FormField.js:237]: Interesting and good catch. So, under what circumstances would the theme not have `formField.border`? If the user specifically set it as `undefined`?

[the remark this answers is not in the document]
Comment (ericsoderberghp) [reply src/js/components/FormField/FormField.js:237]: yes

[the remark this answers is not in the document]
Comment (halocline) [reply src/js/components/FormField/FormField.js:237]: Okay, in that case, in addition I think we need to modify L230 from
```
if (themeBorder.position === 'inner') {
```
to
```
if (themeBorder && themeBorder.position === 'inner') {
```

Same for L336 where we are also checking for `themeBorder.position`, correct?

[the remark this answers is not in the document]
Comment (ericsoderberghp) [reply src/js/components/FormField/FormField.js:237]: Hmm... I was able to reproduce it locally without that change. Let me see if I can remember what I tried...

[the remark this answers is not in the document]
Comment (halocline) [reply src/js/components/FormField/FormField.js:237]: Sounds good. Regardless, I'll add in a test for this case.

[the remark this answers is not in the document]
Comment (ericsoderberghp) [reply src/js/components/FormField/FormField.js:237]: good idea! Would be good to have stories for inner, outer, and none variations across components, like Form -> Controlled, but with different FormField themes.

[the remark this answers is not in the document]
Comment (halocline) [reply src/js/components/FormField/FormField.js:185]: @ericsoderberghp I've gone ahead and added a story for FormField border positions where the theme modifies the `border.position` values, as well as incorporated the check you suggested above.
Screen Shot 2020-06-17 at 11 12 07 AM

**Pull request merged**